service/history/ndc/new_workflow_transaction_manager.go

```diff
@@ -227,10 +237,20 @@ func (r *transactionManagerForNewWorkflowImpl) createAsZombie(
 		return err
 	}
 
-	targetWorkflowHistorySize, err := targetWorkflow.GetContext().PersistFirstWorkflowEvents(
-		ctx,
-		targetWorkflowEventsSeq[0],
-	)
+	var targetWorkflowHistorySize int64
+	if len(targetWorkflowEventsSeq[0].Events) > 0 {
+		if targetWorkflowEventsSeq[0].Events[0].GetEventType() == types.EventTypeWorkflowExecutionStarted {
+			targetWorkflowHistorySize, err = targetWorkflow.GetContext().PersistStartWorkflowBatchEvents(
+				ctx,
+				targetWorkflowEventsSeq[0],
+			)
+		} else { // reset workflows fall into else branch
+			targetWorkflowHistorySize, err = targetWorkflow.GetContext().PersistNonStartWorkflowBatchEvents(
```

Comment:
Do you know where does ndc create the branch record in history_tree table? 
On active side, it calls `ForkHistoryBranch` make that. 
https://github.com/uber/cadence/blob/44392dc5c7c4e0866c0cd73dc43709d5762532fb/service/history/reset/resetter.go#L421

Without that record, the branch info will be corrupted.

Reply:
it's created in this function https://github.com/uber/cadence/blob/master/service/history/ndc/history_replicator.go#L275

Reply:
I may not have the full context here. But the createAs... is for startWorkflowBatchEvents. Is there a bug in reset to call this function with non start batch?